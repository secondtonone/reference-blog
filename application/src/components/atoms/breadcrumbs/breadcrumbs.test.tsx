import renderer from 'react-test-renderer';
import Breadcrumbs from '.';

describe('Match snapshots Breadcrumbs', () => {
  it('empty', () => {
    const tree = renderer.create(<Breadcrumbs />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('with seo page link', () => {
    const tree = renderer
      .create(
        <Breadcrumbs links={
          [
            {
              url: '/',
              name: 'Main page /'
            },
            {
              url: 'seo/',
              name: 'Seo'
            }
          ]
        }
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('with links', () => {
    const tree = renderer
      .create(
        <Breadcrumbs links={
            [
              {
                url: '/',
                name: 'Main page /'
              },
              {
                url: 'seo/',
                name: 'Seo'
              },
              {
                url: 'link/',
                name: 'Link'
              }
            ]
          }
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('with empty url /', () => {
    const tree = renderer
      .create(
        <Breadcrumbs links={
            [
              {
                url: '/',
                name: 'Main page /'
              },
              {
                url: 'seo',
                name: 'Seo'
              },
              {
                url: 'link',
                name: 'Link'
              }
            ]
          }
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
