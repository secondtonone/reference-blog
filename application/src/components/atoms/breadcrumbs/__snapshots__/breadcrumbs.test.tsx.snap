// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshots Breadcrumbs empty 1`] = `null`;

exports[`Match snapshots Breadcrumbs with empty url / 1`] = `
<div
  className="box-adaptive__BoxStyled-sc-g2wy39-0 gfnULt breadcrumbs__BreadcrumbsStyled-sc-1v67b40-0 buCJEi"
  data-test="breadcrumbs"
>
  <ul
    className="text-content__TextStyled-sc-1stkrsz-0 kAbtGH"
    data-test="breadcrumbs-links"
  >
    <li
      data-test="breadcrumbs-link-level-0"
    >
      <a
        href="/"
      >
        Main page /
      </a>
    </li>
    <li
      data-test="breadcrumbs-link-level-1"
    >
      <a
        href="seo/"
      >
        Seo
      </a>
    </li>
    <li
      data-test="breadcrumbs-link-level-2"
    >
      <a
        href="link/"
      >
        Link
      </a>
    </li>
  </ul>
</div>
`;

exports[`Match snapshots Breadcrumbs with links 1`] = `
<div
  className="box-adaptive__BoxStyled-sc-g2wy39-0 gfnULt breadcrumbs__BreadcrumbsStyled-sc-1v67b40-0 buCJEi"
  data-test="breadcrumbs"
>
  <ul
    className="text-content__TextStyled-sc-1stkrsz-0 kAbtGH"
    data-test="breadcrumbs-links"
  >
    <li
      data-test="breadcrumbs-link-level-0"
    >
      <a
        href="/"
      >
        Main page /
      </a>
    </li>
    <li
      data-test="breadcrumbs-link-level-1"
    >
      <a
        href="seo/"
      >
        Seo
      </a>
    </li>
    <li
      data-test="breadcrumbs-link-level-2"
    >
      <a
        href="link/"
      >
        Link
      </a>
    </li>
  </ul>
</div>
`;

exports[`Match snapshots Breadcrumbs with seo page link 1`] = `
<div
  className="box-adaptive__BoxStyled-sc-g2wy39-0 gfnULt breadcrumbs__BreadcrumbsStyled-sc-1v67b40-0 buCJEi"
  data-test="breadcrumbs"
>
  <ul
    className="text-content__TextStyled-sc-1stkrsz-0 kAbtGH"
    data-test="breadcrumbs-links"
  >
    <li
      data-test="breadcrumbs-link-level-0"
    >
      <a
        href="/"
      >
        Main page /
      </a>
    </li>
    <li
      data-test="breadcrumbs-link-level-1"
    >
      <a
        href="seo/"
      >
        Seo
      </a>
    </li>
  </ul>
</div>
`;
